src/optimizer.cpp

```diff
@@ -270,16 +263,37 @@ auto Optimizer::getControlFromSequence(const unsigned int offset) {
   return xt::view(control_sequence_.data, offset);
 }
 
-MotionModel Optimizer::getMotionModel() const { return motion_model_t_; }
+void Optimizer::setMotionModel(const std::string & model)
+{
+  if (model == "DiffDrive") {
+    motion_model_ = std::make_unique<DiffDriveMotionModel>();
+  } else if (model == "Omni") {
+    motion_model_ = std::make_unique<OmniMotionModel>();
+  } else if (model == "Ackermann") {
+    motion_model_ = std::make_unique<AckermannMotionModel>();
+  } else {
+    throw std::runtime_error(std::string(
+        "Model %s is not valid! Valid options are DiffDrive, Omni, or Ackermann", model.c_str()));
+  }
 
-void Optimizer::setMotionModel(const MotionModel motion_model) {
-  motion_model_t_ = motion_model;
-  state_.idx.setLayout(motion_model);
-  control_sequence_.idx.setLayout(motion_model);
+  state_.idx.setLayout(motion_model_->isHolonomic());
+  control_sequence_.idx.setLayout(motion_model_->isHolonomic());
 }
 
 xt::xtensor<double, 3>& Optimizer::getGeneratedTrajectories() {
   return generated_trajectories_;
 }
 
+void Optimizer::setVelocityConstraints(const utils::VelocityConstraints & constraints)
+{
+  vx_max_ = constraints.vx;
```

Comment:
can we make vx/y/z_max then also struct  to use assignment ?

Reply:
Z would imply Z velocity (upward), W is the general velocity term used for angular velocity

Reply:
Yeah sure, it was a typo. 
I mean to create constrsints_ member variable here. I can propose something like that in other PR